### PR TITLE
Updated project to use URLs instead of String paths

### LINKED ISSUE #35: When a path contains '.' or '..' the violation does not show  properly in Xcode
- For showing the errors on Xcode, the printed paths should be absolute and not contain `.` or `..` paths.
- The `directory` parameter in the `violations(in directory: URL)` is not being used in most cases. The reason for such thing is that the `URL(fileURLWithPath: relativePath)` function is used to convert a `String` to a `URL`. Such initializer, when the path is relative, automatically (and silently) sets the base directory to `currentDirectoryPath`, leading to an absolute path equal to:

    ```
    URL(fileURLWithPath: relativePath) = currentDirectoryPath + relativePath
    ```

    Seems that this is wrong: the path should be relative to the `directory` parameter. Thus, function `URL(fileURLWithPath: relativePath, relativeTo: directory)` should be used (only available from `.macOS(.v10_11)`).

    The reason this problem has not been detected in the tests is because the `path` property in the `Resource` object was always returning an absolute path.
    The reason this problem has not been detected in production is because the `currentDirectoryPath` is the directory where the `.projlint.yml` lives, as there is no possibility to specify a custom path for it.

### PR DESCRIPTION
**Pull request by acecilia**
Fixes https://github.com/JamitLabs/ProjLint/issues/35

The main changes in this PR are:
- `url.path` is used now, which returns absolute paths without `.` or `..`.
- Function `URL(fileURLWithPath: relativePath, relativeTo: directory)` is used now when converting a string to URL. Also, the tests are fixed, and now the path returned for a resource is relative instead of absolute.

### CODE REVIEW

**Comment by Jeehut** [comment]
Looks good to me. Will do some tests sometime later and merge if all looks well. Thanks for the PR!

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,6 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
 ### Removed
 - None.
 ### Fixed
-- None.
+- Updated project to use URLs instead of String paths. https://github.com/JamitLabs/ProjLint/pull/33
```
Comment (Jeehut) [inline CHANGELOG.md]: Could you please update this to the format of other entries in the JamitLabs account? Like [here](https://raw.githubusercontent.com/JamitLabs/Accio/stable/CHANGELOG.md)? Thanks!

**Review by Jeehut — COMMENTED** [review]
I think this can be merged. Just one small detail. :)

File: CHANGELOG.md (hunk 1 above)
Comment (Jeehut) [reply CHANGELOG.md]: Also please note that this does not explain, what bug(s) this fixes. Maybe you wanna create an issue which explains it in detail?

Comment (acecilia) [reply CHANGELOG.md]: Done

Comment (Jeehut) [reply CHANGELOG.md]: Thanks! Will merge then, once the CI succeeds.

**Pull request merged**